File: storage/db-core/src/main/java/com/coffee/gu/menu/MenuOptionRepositoryImpl.java
package com.coffee.gu.menu;

import com.coffee.gu.enums.EntityStatus;
import com.querydsl.jpa.impl.JPAQueryFactory;
import org.springframework.stereotype.Repository;

import java.util.List;

import static com.coffee.gu.menu.QMenuOptionGroupEntity.menuOptionGroupEntity;

@Repository
public class MenuOptionRepositoryImpl implements MenuOptionGroupRepository{

    private final JPAQueryFactory jpaQueryFactory;

    public MenuOptionRepositoryImpl(JPAQueryFactory jpaQueryFactory) {
        this.jpaQueryFactory = jpaQueryFactory;
    }

    @Override
    public List<MenuOptionGroup> findByMenuId(Long menuId) {
        return jpaQueryFactory.selectFrom(menuOptionGroupEntity)
                .where(
                        menuOptionGroupEntity.menuId.eq(menuId),
                        menuOptionGroupEntity.entityStatus.eq(EntityStatus.ACTIVE)
                )
                .fetch()
                .stream()
                .map(MenuOptionGroupEntity::toModel)
                .toList();
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/menu/MenuOptionRepositoryImpl.kt
package com.coffee.gu.menu

import com.coffee.gu.enums.EntityStatus
import com.coffee.gu.menu.QMenuOptionGroupEntity.menuOptionGroupEntity
import com.querydsl.jpa.impl.JPAQueryFactory
import org.springframework.stereotype.Repository

@Repository
class MenuOptionRepositoryImpl(
    private val jpaQueryFactory: JPAQueryFactory,
) : MenuOptionGroupRepository {

    override fun findByMenuId(menuId: Long): List<MenuOptionGroup> {
        return jpaQueryFactory.selectFrom(menuOptionGroupEntity)
            .where(
                menuOptionGroupEntity.menuId.eq(menuId),
                menuOptionGroupEntity.entityStatus.eq(EntityStatus.ACTIVE),
            )
            .fetch()
            .map { it.toModel() }
    }
}
